oci/defaults.go

```diff
@@ -154,6 +154,10 @@ func DefaultLinuxSpec() specs.Spec {
 			{Type: "pid"},
 			{Type: "ipc"},
 		},
+		// Always use MS_PRIVATE for rootfs propagation to make sure that we
+		// don't leak any MS_UNBINDABLE mounts into the container's namespace
+		// while its resolving volume bindmounts.
+		RootfsPropagation: "private",
```

Comment:
@rhvgoyal I had a tenative runC patch that would specifically ignore `rootfsPropagation` would explicitly restore `MS_UNBINDABLE` mountpoints after doing `MS_PRIVATE|MS_REC` (or whatever the option is set to). It didn't appear to work and I discovered that `rootfsPropagation=private` fixed the issue as well, so stopped debugging. That method might work.

But as I mentioned we should still use the `rmdir` method, and potentially do something more clever with `EBUSY` in that case.